packages/sdk/ts/verifyingKeys/setVerifyingKeys.ts

```diff
@@ -63,27 +63,29 @@ export const setVerifyingKeys = async ({
     throw new Error("This poll verifying key is already set in the contract");
   }
 
-  // check if the process messages verifyingKey was already set
-  const processVerifyingKeySignature = generateProcessVerifyingKeySignature(
-    stateTreeDepth,
-    voteOptionTreeDepth,
-    messageBatchSize,
-  );
-
-  if (await verifyingKeysRegistryContract.isProcessVerifyingKeySet(processVerifyingKeySignature, EMode.QV)) {
-    throw new Error("This process verifying key is already set in the contract");
-  }
-
-  // do the same for the tally votes verifyingKey
-  const tallyVerifyingKeySignature = generateTallyVerifyingKeySignature(
-    stateTreeDepth,
-    intStateTreeDepth,
-    voteOptionTreeDepth,
-  );
-
-  if (await verifyingKeysRegistryContract.isTallyVerifyingKeySet(tallyVerifyingKeySignature, mode)) {
-    throw new Error("This tally verifying key is already set in the contract");
-  }
+  modes.forEach(async (mode) => {
+    // check if the process messages verifyingKey was already set
+    const processVerifyingKeySignature = generateProcessVerifyingKeySignature(
+      stateTreeDepth,
+      voteOptionTreeDepth,
+      messageBatchSize,
+    );
+
+    if (await verifyingKeysRegistryContract.isProcessVerifyingKeySet(processVerifyingKeySignature, mode)) {
+      throw new Error("This process verifying key is already set in the contract");
+    }
+
+    // do the same for the tally votes verifyingKey
+    const tallyVerifyingKeySignature = generateTallyVerifyingKeySignature(
+      stateTreeDepth,
+      intStateTreeDepth,
+      voteOptionTreeDepth,
+    );
+
+    if (await verifyingKeysRegistryContract.isTallyVerifyingKeySet(tallyVerifyingKeySignature, mode)) {
```

Comment:
Let's move it to a separate variable

Reply:
I am removing these changes for a new PR later